### PR TITLE
PA-1210 Parcel Detail Form redesign with test corrections.

### PR DESCRIPTION
**Pull request by devinleighsmith**
Known Issues:
No tooltips on building page numbers. Delaying this due to time constraints. The library we are using doesn't give direct access to the page number component so adding the tooltips may be tricky.
Button inconsistency - waiting for Bobbi/Alex to meet before I update.
Pins don't always display below popups when jumping to a lat/lon. This is actually an existing issue that doesn't fit within the scope of this story.
PIN move warning is no longer displayed. This is intentional as it seems pointless to re-add this when I have a story this sprint to add a move pin button for parcels and buildings.

### CODE REVIEW

File: frontend/src/components/common/TooltipWrapper.tsx [hunk 1]
```diff
@@ -13,9 +13,16 @@ interface ITooltipWrapperProps extends Partial<OverlayTriggerProps> {
 const TooltipWrapper: React.FunctionComponent<ITooltipWrapperProps> = props => {
   return (
     <>
-      <OverlayTrigger {...props} overlay={<Tooltip id={props.toolTipId}>{props.toolTip}</Tooltip>}>
-        {props.children}
-      </OverlayTrigger>
+      {props.toolTip?.length ? (
```
Comment (devinleighsmith) [inline frontend/src/components/common/TooltipWrapper.tsx:16]: no point in displaying the overlay if there is no tooltip text.

File: frontend/src/components/common/form/FastCurrencyInput.tsx [hunk 2]
```diff
@@ -94,7 +94,7 @@ const CurrencyInput = ({
           onBlur={handleBlur}
           className={classNames('form-control', className, isInvalid, isValid)}
           disabled={disabled}
-          placeholder={placeholder || '$0'}
+          placeholder={placeholder || ''}
```
Comment (devinleighsmith) [inline frontend/src/components/common/form/FastCurrencyInput.tsx:97]: stop making $0 the placeholder as it is confusing to users.

File: frontend/src/components/maps/leaflet/Map.tsx [hunk 3]
```diff
@@ -309,100 +315,107 @@ const Map: React.FC<MapProps> = ({
...
-                autoPan={false}
-              >
-                <LayerPopupContent
-                  data={layerPopup.data as any}
-                  config={layerPopup.config as any}
-                />
-              </Popup>
-            )}
-            <LegendControl />
+    <ReactResizeDetector handleWidth>
```
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/Map.tsx:318]: Get the map to react to the component being resized.

File: frontend/src/customAxios.ts [hunk 4]
```diff
@@ -50,9 +50,11 @@ const CustomAxios = ({
 
   instance.interceptors.response.use(
     response => {
-      if (lifecycleToasts?.successToast) {
+      if (lifecycleToasts?.successToast && response.status < 300) {
```
Comment (devinleighsmith) [inline frontend/src/customAxios.ts:53]: some errors do not cause a promise rejection, make sure to not treat those as successes.

File: frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import MapSideBarLayout from '../components/MapSideBarLayout';
+import useParamSideBar from '../hooks/useQueryParamSideBar';
+import ParcelDetailContainer from 'features/mapSideBar/containers/ParcelDetailContainer';
+
+interface IMapSideBarContainerProps {}
+
```
Comment (goemen) [inline frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx:7]: No need  for an empty interface

File: frontend/src/features/properties/submit/SubmitProperty.tsx [hunk 6]
```diff
@@ -179,16 +177,7 @@ const SubmitProperty = (props: any) => {
...
-              <ParcelDetailForm
-                agencyId={keycloak.agencyId}
-                clickLatLng={leafletMouseEvent?.latlng}
-                parcelDetail={parcelId ? (cachedParcelDetail as IParcel) : null}
-                disabled={formDisabled || readonly}
-                secret={keycloak.obj.subject}
-                loadDraft={!!loadDraft}
-              />
-            </Col>
+            <Col></Col>
```
Comment (Fosol) [inline frontend/src/features/properties/submit/SubmitProperty.tsx:180]: Is this waiting for the edit form?

**Review by Fosol — APPROVED** [review]
Looking forward to seeing this

File: frontend/src/features/mapSideBar/containers/ParcelDetailContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    formikRef.current.setFieldValue('latitude', leafletMouseEvent.latlng.lat);
+    formikRef.current.setFieldValue('longitude', leafletMouseEvent.latlng.lng);
+  }
+  return (
+    <ParcelDetailForm
+      formikRef={formikRef}
+      currentTab={currentTab}
+      setCurrentTab={setCurrentTab}
+      parcelDetail={null}
+      secret="blah"
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/containers/ParcelDetailContainer.tsx:36]: This will be updated for the view/update logic.

**Review by jdtoombs — APPROVED** [review]

**Review by goemen — APPROVED** [review]

File: frontend/src/components/common/form/Input.tsx [hunk 8]
```diff
@@ -94,27 +99,32 @@ export const Input: React.FC<InputProps> = ({
...
-          }
-          handleBlur(e);
-        }}
-        className={className}
-        onChange={pattern ? handleRestrictedChange : handleChange}
-      />
+      {!!tooltip && !label && <TooltipIcon toolTipId={`${field}-tooltip`} toolTip={tooltip} />}
+
+      <TooltipWrapper toolTipId={`${field}-error-tooltip}`} toolTip={errorTooltip}>
+        <Form.Control
```
[the remark this answers is not in the document]
Comment (devinleighsmith) [reply frontend/src/components/common/form/Input.tsx:105]: Every element has a tooltip that is only visible if there is formik error text.

File: frontend/src/components/maps/BuildingPopupView.tsx [hunk 9]
```diff
@@ -75,9 +76,23 @@ export const BuildingPopupView: React.FC<IBuildingDetailProps> = (props: IBuildi
             {!props?.disabled &&
               (!keycloak.hasAgency(buildingDetail?.agencyId as number) &&
               !keycloak.hasClaim(Claims.ADMIN_PROPERTIES) ? (
-                <Link to={`/submitProperty/${buildingDetail?.parcelId}?disabled=true`}>View</Link>
+                <Link
+                  to={{
+                    pathname: `/mapview/${buildingDetail?.parcelId}?disabled=true`,
+                    search: location.search,
```
[the remark this answers is not in the document]
Comment (devinleighsmith) [reply frontend/src/components/maps/BuildingPopupView.tsx]: Not at the time, as I was not implementing view/edit.

File: frontend/src/features/mapSideBar/components/MapSideBarLayout.tsx [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+ */
+const MapSideBarLayout: React.FunctionComponent<IMapSideBarLayoutProps> = ({
+  show,
+  setShowSideBar,
+  ...props
+}) => {
+  return (
+    <div className={classNames('map-side-drawer', show ? 'show' : null)}>
+      <HeaderRow>
+        <h2 className="mr-auto">Property Details</h2>
```
[the remark this answers is not in the document]
Comment (devinleighsmith) [reply frontend/src/features/mapSideBar/components/MapSideBarLayout.tsx]: I'm of the school of thought that over-genericizing components that are only ever meant to be used in one way is an anti-pattern. My preference is to create non-generic components in most instances and then update those to be generic if there is a proven need in the application.

File: frontend/src/features/mapSideBar/containers/ParcelDetailContainer.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (devinleighsmith) [reply frontend/src/features/mapSideBar/containers/ParcelDetailContainer.tsx]: the secret is used to store the form data in localstorage.

File: frontend/src/features/properties/submit/SubmitProperty.tsx (hunk 6 above)
Comment (devinleighsmith) [reply frontend/src/features/properties/submit/SubmitProperty.tsx]: Yep, deleted now.

**Comment by devinleighsmith** [comment]
Currently dealing with adminarea conflicts.

**Comment by devinleighsmith** [comment]
SNYK!!!!

**Pull request merged**